1693 report import (#1987)
ICTU/quality-time: docs/USAGE.md

To export an older version of a report, add the `report_date` parameter with a date value in ISO-format (YYYY-MM-DD), for example `http://www.quality-time.example.org/api/v3/report/<report_uuid>/pdf?report_date=2020-09-01`.

## JSON report import/export

> *Quality-time* provides functionality for importing and exporting reports in JSON format. This functionality can be used for backing up reports or for transferring reports from one *Quality-time* instance to another one. Currently this functionality is only available via the api, with one endpoint for importing and one for exporting the JSON reports.

Reviewer: ```suggestion
*Quality-time* provides functionality for importing and exporting reports in JSON format. This functionality can be used for backing up reports or for transferring reports from one *Quality-time* instance to another one. Currently this functionality is only available via the API, with one endpoint for importing and one for exporting the JSON reports.
```

A *Quality-time* report JSON contains the configuration of the report, with all it's subjects, metrics and sources. It does not contain any actual measurements. 
The credentials of configured sources will be encrypted to prevent the loss of sensitive data.

### Export API

> The exporting endpoint is available under `http://www.quality-time.example.org/api/v3/report/<report-uuid>/json?public_key=<public-key>`. The public key argument is optional, if no public key is provided, the public key of the exporting *Quality-time* instance is used for encrypting the source credentials. If the report needs to be imported in a different *Quality-time* instance, the public key of that instance should be provided. It can be opbtained at `www.quality-time.example.org/public_key`. The exported JSON report can only be imported into the *Quality-time* who's public key has been used for the credential encryption during the export.

Reviewer: Is that really the correct URL for the public key?

```suggestion
The exporting endpoint is available under `http://www.quality-time.example.org/api/v3/report/<report-uuid>/json?public_key=<public-key>`. The public key argument is optional. If no public key is provided, the public key of the exporting *Quality-time* instance is used for encrypting the source credentials. If the report needs to be imported in a different *Quality-time* instance, the public key of that instance should be provided. It can be obtained at `www.quality-time.example.org/api/v3/public_key`. The exported JSON report can only be imported into the *Quality-time* whose public key has been used for the encryption of credentials during the export.
```

### Import API

Exported reports can be imported back into *Quality-time*. The endpoint for this is `http://www.quality-time.example.org/api/v3/report/import`. The import accepts JSON content only.

> Only exports that have been exported using the public key of the imported *Quality-time* instance can be imported. The Source credentials will be decrypted on import and the sources should be ready to collect measurements automatically.

Reviewer: ```suggestion
Only exports that have been exported using the public key of the destination *Quality-time* instance can be imported. The source credentials will be decrypted on import and the sources should be ready to collect measurements automatically.
```
